src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java

```diff
@@ -169,14 +168,15 @@ public Stream<SourceFile> listSourceFiles(MavenProject mavenProject, @Nullable X
             alreadyParsed.add(baseDir.resolve(maven.getSourcePath()));
         }
 
-        Object mavenSourceEncoding = mavenProject.getProperties().get("project.build.sourceEncoding");
-        if (mavenSourceEncoding != null) {
-            ParsingExecutionContextView.view(ctx).setCharset(Charset.forName(mavenSourceEncoding.toString()));
-        }
         JavaParser.Builder<? extends JavaParser, ?> javaParserBuilder = JavaParser.fromJavaVersion()
                 .styles(styles)
                 .logCompilationWarningsAndErrors(false);
 
+        Object mavenSourceEncoding = mavenProject.getProperties().get("project.build.sourceEncoding");
```

Comment:
Ideally, we would also accept the encoding from the compiler plugin configuration: https://www.baeldung.com/maven-encoding

Reply:
done in https://github.com/openrewrite/rewrite-maven-plugin/pull/735/commits/175d711c6a1febed5e8734adfc11b31bfc60cfee , thanks!

Reply:
Were you able to check if the precedence is correct between the plugin configuration and the project property?